web/src/main.js

```diff
@@ -26,10 +28,12 @@ Vue.use(VueRouter)
 // configure the app's routing
 import Home from '@/pages/Home.vue'
 import Project from '@/pages/Project.vue'
+import Docs from '@/pages/Docs.vue'
 
 const routes = [
   { path: '/', component: Home },
-  { path: '/:project', component: Project }
+  { path: '/:project', component: Project },
+  { path: '/:project/:version/:location?', component: Docs }
```

Comment:
the `?` is for with trailing slash or not correct? if so would it make sense to do the same for /:project

Reply:
Nope it is for an optional parameter: https://stackoverflow.com/a/47828416